### PR TITLE
Identity lookup

### LINKED ISSUE #44: DeltaApi - Identity Lookup by metadata

### PR DESCRIPTION
**Pull request by myuwono**
closes #44

### CODE REVIEW

File: src/test/python/test_apiclient.py [hunk 1]
```diff
@@ -324,6 +325,40 @@ def test_get_secret_content(api_client, mock_signer):
...
+
+    mock_signer.assert_called_once_with(requestor_id)
+
+    assert len(responses.calls) == 1
+    assert response == expected_json
+    url = urllib.parse.urlparse(responses.calls[0].request.url)
+    query_params = dict(urllib.parse.parse_qsl(url.query))
+    expected_query_params = {
+        "metadata.name": "test123",
+        "page": "1",
```
Comment (numbat) [inline src/test/python/test_apiclient.py]: Do we want a few tests for when page or pageSize is missing or invalid?

Comment (myuwono) [inline src/test/python/test_apiclient.py]: good point, yes we do

File: src/main/python/covata/delta/utils.py [hunk 2]
```diff
@@ -15,30 +15,52 @@
...
+
+
+def check_arguments(arguments, test_type, test_function, fail_message):
+    def decorator(function):
+        def _f(*args, **kwargs):
+            for arg, value in kwargs.items():
+                if arg in arguments:
+                    if not any(value is t or isinstance(value, t)
+                               for t in test_type):
+                        raise TypeError(
```
Comment (myuwono) [inline src/main/python/covata/delta/utils.py]: not sure if raising `TypeError` is pythonic. Enforcing exact type (i.e. bear-typing) goes against duck-typing principles.

File: src/main/python/covata/delta/utils.py [hunk 3]
```diff
@@ -15,30 +15,52 @@
...
+        caller_frame.f_locals['self'].__class__.__name__
+        if 'self' in caller_frame.f_locals else None])
+    del caller_frame
+    return ".".join(name)
+
+
+def check_arguments(arguments, test_type, test_function, fail_message):
+    def decorator(function):
+        def _f(*args, **kwargs):
+            for arg, value in kwargs.items():
```
Comment (myuwono) [inline src/main/python/covata/delta/utils.py]: this will not work when `kwargs` is empty, i.e. user passes in `args` or partial `kwargs`

**Pull request merged**